internal/sorter/sorter.go

```diff
@@ -119,6 +119,24 @@ func (s routeSorter) Less(i, j int) bool {
 		case *envoy_route_v3.RouteMatch_Prefix:
 			return true
 		}
+	case *envoy_route_v3.RouteMatch_Path:
```

Comment:
Now that prefix matches have become regexes, could this have some weird unintended ordering/precedence consequences? Especially since we add a few characters in the prefix regex, and we sort regex matches on length? Haven't really thought about it yet

Reply:
Potential solution:
- Prefix "regex" matches stay as a Prefix match type with a flag telling that they are not string prefix
- Sort routes before converting to Envoy types